rules/typescript.js

```diff
@@ -240,6 +240,8 @@ module.exports = {
      * Superseded by @typescript-eslint/no-unused-vars.
      */
     '@typescript-eslint/no-unused-vars-experimental': 'off',
+    // Conflicts with non-null assertions.
+    '@typescript-eslint/non-nullable-type-assertion-style': 'off',
```

Comment:
This is actually correct. It changes things like `value as string` to `value!`, which is prohibited - making it clear for developer that this is dangerous in the same way (there's no real difference between the two, just a different syntax). We had a loophole before where developers were simply using `as X` when it was not safe to do so.